src/fs.cpp

```diff
@@ -90,6 +90,122 @@ bool FileLock::TryLock()
     }
     return true;
 }
+
+#endif
+
+#if defined WIN32 && defined __GLIBCXX__
+
+/**
+*  reference: https://github.com/gcc-mirror/gcc/blob/571ee70a6d4226145f77b885937a8eda678e0fb2/libstdc%2B%2B-v3/include/std/fstream#L283
+*
+*  +---------------------------------------------------------+
```

Comment:
I think I would drop this table but keep the URL link, since the table just duplicates information provided by the code below, and seems actually less readable than the code.

For reference, https://en.cppreference.com/w/cpp/io/basic_filebuf/open also has a similar table.